### PR TITLE
freshenv 0.1.2 (new formula)

### PR DESCRIPTION
**Pull request by raiyanyahya**
freshenv is a command line application to provision and manage local developer environments.

I've made this formula available as a tap which can be installed like so:
brew install raiyanyahya/freshenv/freshenv

See https://github.com/raiyanyahya/freshenv for more details.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/freshenv.rb [hunk 1]
```diff
@@ -0,0 +1,89 @@
+class Freshenv < Formula
+  include Language::Python::Virtualenv
+
+  desc "Provision and manage local developer environments"
+  homepage "https://github.com/raiyanyahya/freshenv"
+  url "https://files.pythonhosted.org/packages/88/91/75539ca1a7b0aec86dc5759be1a6167edc017e950beecd1d67f3018587d6/freshenv-0.1.2.tar.gz"
+  sha256 "9cbed9143cb904b5fa5c2f226ecd95a9f9b5692cee773b85cfb402020176d273"
+  license "MIT"
+  depends_on "python@3.8"
```
Comment (carlocab) [inline Formula/freshenv.rb:9]:
```suggestion
  license "MIT"

  depends_on "python@3.8"
```
Does this not work with a newer Python?

File: Formula/freshenv.rb [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+  resource "rich" do
+    url "https://files.pythonhosted.org/packages/b3/3b/5a90d0f197087896472757ebd9ac29d802ccc4f6953d99a535b1a8db2bf6/rich-10.16.1.tar.gz"
+    sha256 "4949e73de321784ef6664ebbc854ac82b20ff60b2865097b93f3b9b41e30da27"
+  end
+
+  resource "six" do
+    url "https://files.pythonhosted.org/packages/71/39/171f1c67cd00715f190ba0b100d606d440a28c93c7714febeca8b79af85e/six-1.16.0.tar.gz"
+    sha256 "1e61c37477a1626458e36f7b1d82aa5c9b094fa4802892072e49de9c60c4c926"
+  end
```
Comment (carlocab) [inline Formula/freshenv.rb:69]: This should do `depends_on "six"` instead of declaring a resource.

File: Formula/freshenv.rb [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    sha256 "4987c65554f7a2dbf30c18fd48778ef124af6fab771a377103da0585e2336ece"
+  end
+
+  resource "websocket-client" do
+    url "https://files.pythonhosted.org/packages/b6/fa/72e77d094563208174abbbaa73c32f28c43a31193b843bddf233c7c87644/websocket-client-1.2.3.tar.gz"
+    sha256 "1315816c0acc508997eb3ae03b9d3ff619c9d12d544c9a9b553704b1cc4f6af5"
+  end
+
+  def install
+    virtualenv_create(libexec, "python3")
```
Comment (carlocab) [inline Formula/freshenv.rb:82]:
```suggestion
```
This isn't needed.

File: Formula/freshenv.rb [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    virtualenv_install_with_resources
+  end
+
+  test do
+    system bin/"freshenv", "--help"
```
Comment (carlocab) [inline Formula/freshenv.rb:87]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

**Review by carlocab — COMMENTED** [review]
Thanks for the PR, @raiyanyahya.

This looks like a nice piece of software, but it currently does not meet our notability requirements. (See the output of `brew audit --new freshenv` when you have your PR branch checked out.)

If you wish, you can host this formula in your own tap instead. Here are some docs to help you get started:

https://docs.brew.sh/Taps
https://docs.brew.sh/Interesting-Taps-and-Forks
https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap
https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/

If you choose to re-submit this PR in the future, please address my comments below before doing so.

**Comment by raiyanyahya** [comment]
> please address my comments below before doing so.

@carlocab Thank you so much for the reply and the review. I will start working on them immediately. As far as notable repos are concerned I am afraid I do not think I will be able to get so many stars so soon. I understand that you require repos to have some credibility and use else everyone will start pushing random packages on homebrew. Is their any way I can get this on the core brew without being notable ? I already have a self hosted tap, but I wanted a premium experience for developers when I launch this product (Mac users would prefer it, and it gives me some credibility too I guess)
Anyways I will start working on the changes you proposed. Thank you for your time :).

**Review by raiyanyahya — COMMENTED** [review]
Thank you for time with the review. Will push a change with your review comments addressed.

**Pull request closed without merging**